Configurations and Updates.txt
- Downloaded Source/Client (Divine something?)
- Started debugging and looking for bugs threwout client/source
- Started some client issues (Interface interfearing and such somehow)
- Made an attempt on fixing firecape model issue (it was only going on half the back v-Pic below)
. But instead it became white when i packed a newer firecape model (but this one is wearing as it should)
- Messing around with different character models
- Looked threw some content on the source and decided to keep some (as the pvp aspect, v-Pic below)
- Removed Quick curses completely it was totally messed up (might be intergrated later on)
- Fixed some special attack emotes as they were totally messed up and wrong in every way
- Dumped NPC Emote ID's and corrected loads of NPC emotes (not tested all yet)
- Started the Fishing Colony as one main "skilling spot" aswell as catherby
- Added client commands ::x10on/::x10off for toggling x10 hits on and off
- Moved home to edgeville as usual
- Added a banker and a skilling supply shop at the Fishing Colony along with some thiev stalls
- Added prayer/summoning restorers in edge and also prayer book alter switch
- Finished the teleportation system spellbook wise
- Created the skill master who originally was Bob and placed him at the Fishing Colony and made him a teleporter aswell to various skilling areas
- Patched a dupe
- Fixed choosing of dicebags
- Emptied out some unused teleports for now
- Fixed the Blow kiss emote with the belonging gfx
- Revamped and fixed up some staff commands
- Fixed Intermediate Mining location teleport
- Fixed the rocks in the crafting guild
- Some item pricing has been redone
- Fixed Pernix Cowl somewhat as it should be, (client sided issue)
- Corrected some armoury wears such as fullhelms, fullbodys etc..
- Added Bronze Set (lg) to shop
- Fixed Black Set (sk) to be opened
- Removed Wilderness agility course till further notice due to object missing
- Fixed skillcape and slayer shop
- Fixed slayer points loading
- More armour wielding fixes
- Fixed Strength requirements for weapons
- Fixed strykewyrms distance attack
- Fixed back worldmap icon
- Fixed up jailing system
- Completed Easter ring and ring of stone
- Added special bar for some weapons that we're missing it
- Added an achievement system for people to complete
- Packed the old lunar interface without miasmic, don't want them
- Zipped cache with new sprites
__________________ SERVER ONLINE ___________________________ NOVEMBER 24th 2013 ______
24/11-13:
The animation when runecrafting by altars is fixed properly
The teleport to the blood altar has been fixed
Dragon attack/block/death emotes has been fixed
Added a miniquest The Strykewyrm, required to slay Strykewyrms
Glory now also teleports to karamja
The Pocketeer achievement now grants 60K Thieving xp when finished
Skilling experiences has been increased slightly
Zamorak boss has been fixed properly

25/11-13:
A chamber guardian has been added to the mage arena to keep count on Mage points
Ring and Necklace mould has been added to the skilling shop for jewelry making
Slayer points now show as they should in the statistics menu
Finished voting system by GTL
Crystal key chest is now placed outside edgeville bank
All bird's nests are fully functional giving items
Membership system is now fully functional which constists of 31 Days
Random npc spawns has been removed
Added a timer for clicking mining ores
Added a timer for clicking woodcutting trees
Finished the new teleportation system which is handled by the sailor in edgeville
Added the Cave Of Mutations

26/11-13:
The spellbook teleports has been removed
Membership options has been added
Added global announcement when someone reaches level 99 in a skill
Chaos Dungeon added
Over 15 Npc's added to the slayer task lists
Membership can now be purchased
Member only area has been started on

27/11-13:
Fixed some doors in lunar isle
Fixed the magebook loading on login
Fixed Godwars dungeon teleport location
Fixed Frost dragons teleport location

28/11-13:
Finished Lunar Diplomacy quest - http://gyazo.com/8ec5903e04cd7bf27ad7b2abe5f7d327.gif
http://gyazo.com/eac219d7749e559d3312c938161f074c
Added lunar altar north of edgeville bank
Added bowstrings to the skill master's shop
Moved shops to the general shop of edgeville to keep cluster out
Fixed up some item prices
Added some more global doors
Added 8 new achievements
Godwars bosses only requires 20 KC now instead of 30

29/11-13:
An issue regarding dfs on npc gfx has been fixed
An sql error has been fixed
ObjectID showing for members is now removed
New client update, achievement interface has been reworked and
you can now click each achievement to see your progress!
The PKing teleport 'The Gates' now brings you outside the gates above KBD lair
Starter money reduced from 5M To 2M
Single and double doors should now work as usual even when changing regions
Alot of price adjustments
Shop supplys have been limited down

30/11-13:
Mith Grapple has been added to wydins weapon shop
You can no longer save teleports in godwars boss chambers
Your remaining membership days now shows on login screen
Regular users now gets 2 saved teleport slots, and members 5
Fixed the update command timer to work at normal speed
Membership users now have theyre very own slayer dungeon
New way to create an onyx amulet:
- Use your cut onyx on a amulet mould with 1 gold bar in your inventory.
Increased fishing XP slightly
Creating a onyx amulet now requires 90 Crafting
Adjusted alot of droprates
Global objects loading should be working better now
By using a whip vine on a regular whip you will now get a Vine Whip
Fixed up Kree'arras size due to graphic bugs
Added the correct attack,block and death emote to Kree'arra
Fixed up Flambeeds size due to graphic bug
Added the correct attack,block and death emote to Flambeed
Added correct attack,block and death emotes to many other npcs
Added magic attacks to some mages in the chaos tunnels
Client updated to V1.1
Fixed a combat bug

2/12-13:
Rewrote the membership system and it should now calculate properly.
Added some more user icons (donator, server informator, veteran etc!)
A message has been added when claiming barrows reward with less inventory space then needed that the items are put in the bank.
Fixed up the 525 gameframe some more
Fixed up the equipment tab to an older version

3/12-13:
Adjusted fishing spot sizes (were 4x4 click region)
Fixed user icons that showed wrong icon when toggling private chat to show in chatbox.
You can now click the summoning tab again (when having a familiar spawned)
Patched another dupe before anyone got a hold of it
Completely rewritten the crystal chest loot within 3 categories
Common
Un-common 
Rare
Client now updated to V2.1 (Reload webclient/download latest client to avoid ingame issues)

4/12-13:
Membership system has been edited some more
Regular users can now fish and cook rocktails at level 99 fishing and cooking
Added correct map dots for friends, clanchat members and team capes
More rewards added to the crystal chest
Added a cooking range to the skilling area
Fixed up private messaging not showing/sending some messages
Client now updated to V2.2 (Reload webclient/download latest client to avoid ingame issues)

5/12-13:
A few items added to the donator shop
Npcs do no longer drop unnoted items in such huge packs that they crash the client
Finished private messaging bug (sorry for the wait)
Ringmaster set now wields correctly
Increased fishing XP slightly
Increased cooking XP slightly
Finished the 3 different member symbols
Fixed the rocktails fishing in skilling area
Added bank booth at members area

7/12-13:
Santa's Costume Set added to donator shop
Removed cyclops from slayer tasks
Cave bugs in the Chaos Tunnels are now attackable
Added 2 new membership ranks to the shop
Donator points is now introduced (with the new ranks)
Bandos boss now has a higher max hit and had the respawn time decreased

8/12-13:
Wilderness agility course has been added (instead of rocks, use the ladder to fulfill the course)
Nex now drops noted green d'hide instead of unnoted
Dungeoneering beta released
Fixed dungeoneering npc's
Dungeoneering interface should no longer interfear with the logout tab
Prayer should now be recharged after finished wave
Skillcape shop moved to edgeville
Completionist rack moved to edgeville
Mysterious shades in dungeoneering should have theyr correct animations
Donator Points added to the statistics menu
The bank booth in members area now works properly

9/12-13:
Dungeoneering updates:
Floor 1 Complexity 3 is now fixed and can be finishes
When leaving a floor manually the logout interface should not be changed
When teleporting away from the dungeoneering lobby the logout interface should not be changed
Mysterious shades should no longer flip out when attacking (forgot an ID)
Bound mauls has theyr attack emote fixed

Skilling updates:
Creating extremes works now (simple methods for now)
To smelt mithril, adamant and rune bars you now only need 1 Coal

Misc updates:
Warhammers are now wielding in the weapon slot instead of shield spot
The door to the cooking range in catherby now works
The door to the chemist in draynow now works
Big bones was removed from the supply shop
Grenwall spikes has been added to the supply shop
The stock of mithril, adamant and rune bars has been limited down
D'hides in the skilling shop has been replaced by noted ones
Mime gloves added to Zambo's outfit shop
Added a Prestige Point shop to edgeville
Added a Vote Point shop to edgeville
Battleaxes has theyr attack emote fixed
When creating overloads you now receive 1 instead of 2

10/12-13:
Bandos respawn time has been decreased from 4 minutes to 1,5minutes
Fixed vine whips bonuses
The last option from sailor does no longer open up dice menu
Third-age druidic is no longer able to be sold to the shop
Finished the Desert Treasure quest

11/12-13:
Dungeoneering floor 3 has been released to beta
Fixed the completion interface for Desert Treasure
Fixed magic book not showing for new members
Removed iron and mythic crossbows from the range shop
Limited down the stock of items in the range shop
Added drops to the Dagon'Hai monks
Webclient now built into the forums
Store now built into the forums
(:: store) command added which brings up the donation store

12/12-13:
Christmas sale on the web store

13/12-13:
Newcomer tutorial was added for the basics

17/12-13:
Killing streak now ends when you die
The skilling tasks system has now been added to flynn who is located at the skilling area

18/12-13:
New member area (island)
Placed out misc objects threwout the island:
- Magic book switching altars
- Prayer altar
- Spec restore altar
- Summoning restore
- x3 Magic trees
- x4 Rune ores
- x2 Bank booths
- Furnace
- Anvil
- Cooking range
- Donator Chest
Fixed up Donator chest which takes chest chances that can be bought or is included in some purchases!
Filled donator chest with looads of items, some great ones and some less great ones.
Vanguard set was added to the web store
Permanent member ranks was added to the web store
X2 Points was added to the web store (This is a Limited deal!)
Chest chances was added when purchasing a member rank
Silly jester tights now wields properly as legs and not a weapon (silly yeah..)
Santa's costume set (limited deal item)'s price was lowered slightly
X2 Points (limited deal item) price was lowered slightly
Chest Attempts can now be purchased separately
::donor command added for donators

19/12-13:
Fixed up dungeoneering floor 2 complexity 3
Lowe now sells magic shortbows again
Christmas event has been released!
Client updated to Version 2.4 - Christmas edition!
The experience when making extremes has been buffed

27/12-13:
Black tiles fixed (client issue)
Rings creation has been fixed up and should work properly
The - Help - part in quest tab was renamed to - Config - 2 new options below:
Redirects (Opens a tab with fast redirects)
Carried Wealth (Shouts your current Carried Wealth)
The experience when making extremes has been buffed
Most single doors removed (All over the world)
Vote point shop exchange fixed up
Some npc/objects renamed to become cleaner
Client updated to Version 2.5 - Make sure to update!
Members can now buff Godwars killcount +20 (Once per half hour)
Flynn (Skill task manager) Now has a shop for skill points
The sailor is now the 'traveler' and has also been given a Quick travel Option to skip the first dialogue
The vote handling guy was given 2 new options 1- View Rewards 2- Open Vote Page for quick accessing
You can now create onyx amulets threw the interface
The furnace EAST of edgeville bank now works as the Jewelry Making 'Station'-
- Simply click the furnace to open the interface
When attacking a NPC without the correct slayer level, it will now tell which level you need.

03/01-14:
Finished update server (Faster cache files download, (Now only checks and downloads the updated files))
Removal off all doors reworked, now only removing some, including the one for the furnace building in edgeville
Fishing XP for Mantas and Rocktails has been increased
The christmas event is now canceled for 2013! (Abit overdue)
Christmas Webshop Item Deals are now canceled! (Also abit overdue)

08/01-14:
New achievement - XP Master
New achievement - Effigy Locator
New achievement - Effigy Inspector
New achievement - THE Slayer
New achievement - Nex's Destroyer
The summoning area has been moved to a new location
Dice bags are now removed from the web shop and added into Wydins store at a limited price
Zeke was replaced by the Magic store owner
The weapons in Wydins weapon shop is now located on the bottom in Grum's Stock 1
Wydin now holds Miscellanious items instead of weapons
Mithril seeds has been added to Wydins Miscellanious shop
 
12/01-14:
Scimitars of most sorts was moved to grum's store Stock 2
A new tab was created 'Total info' which includes total ingame playtime and total NPC kills

13/01-14:
Melee XP rate has been DOUBLED from 750xp per hit to 1500xp per hit
The melee walk delay in combat should now be removed completely
Abyssal demons now drops whip more reguarly
All godwars bosses and minions droprates has been buffed up alot
The auto cache updater was disabled for now to speed up the cache download speed
Mandrith now holds PK Supply's and buys all of your statuettes
Mandrith has been placed by the wilderness ditch
The snow on the ground has melted and will hopefully be back next year!

14/01-14:
Updated Highscores system - should no longer cause random disconnections
Added beginners tutorial to newly created accounts
Finished runecrafting pouches
The PK Supply shop should no longer take tokkul's instead of coins

15/01-14:
Soulsplit now heals a 4th of your damage made instead of constant 2's
Dharok set boost has been nerfed down abit due to OP hits

23/01-14:
Finished Chaos elemental boss
Dharok's set boost has been buffed up again seeming it was nerfed abit too much
You can now name your saved teleports by using the following:
Example: ::tele1 edgevile (Will name your first saved teleport to Edgeville)
This works for all 5 teleports simply change ::tele1 to ::tele2, ::tele3, ::tele4 or ::tele5
The Salve Amulet has been added to the donator webshop, this amulet keeps your prayer points from draining at all times!


4/02-14:
Chest chances was removed from the web store
Monthly membership has been removed (Those who still have days left will still have it untill it runs out)
Permanent Membership prices and rewards has been adjusted accordingly
* Member: 10$
* Super Member: 50$
* Extreme Member: 100$
Some new sets of perks for each invidual member rank is being worked on
Scroll of imbuedment was removed from the web store
Added Try Your Luck boxes to the web store accordingly:
* Try Your Luck Box [EASY] - Grants item(s) ranging from normal to rare (At a low rate)
* Try Your Luck Box [MEDIUM] - Grants item(s) ranging from normal to rare (At a higher rate)
* Try Your Luck Box [ELITE] - Grants item(s) ranging from normal to rare (At the highest rate)
You can now choose your own experience rate at start with 3 different stages
* Novice 5000xp
* Expert 500xp
* Legend 50xp
Your XP title will now be shown on the overall highscores page

6/02-14:
Novice and Expert modes can no longer use dragonkin lamps (Opinions?)
All XP Modes should now receive the same amount of xp in the dungeoneering floors
All XP Modes should now receive the same amount of xp in the slayer tasks
Novice mode now gets 10 Slayer points upon finished task
Expert mode now gets 15 Slayer points upon finished task
Legend mode now gets 30 Slayer points upon finished task
Client updated to V3.3 (Make sure to refresh webclient or download latest desktop client)
Title Manager has found hes way to edgeville and can provide you to unlock and change titles!
* Titles: [Nov. PK], [Exp. PK], [Leg. PK] added to be unlocked (Requires 500Player Kills)
* Titles: [Nov. PVM], [Exp. PVM], [Leg. PVM] added to be unlocked (Requires 1500 NPC Kills)
The Title Manager's dialogue should now be closing correctly when unlocking or changing title

8/02-14:
Receiving starter after completing beginners tutorial should now work properly
2 New more explaining dialogues has been added for selecting game mode and receiving starter
Novice and Expert modes can now use dragonkin lamps again which grants 500K XP while it grants 1M XP for Legend modes
Everyone can now spend dragonkin lamp xp on dungeoneering

Destroscape New/src/game/Config.java
package game;

import engine.util.Misc;

public class Config {

	/** Debugging/Misc. Config **/
	public static boolean SERVER_DEBUG = false; // Enable's server debugging
												// mode for Developer+
	public static final boolean WORLD_LIST_FIX = false; // change to 'true' if
														// you want to stop that
														// world--8

	/** Server Messaging Config **/
	public static final String SERVER_NAME = "Destroscape";
	public static final String WELCOME_MESSAGE = "Welcome to @red@"+Config.SERVER_NAME+"!";
	public static int MESSAGE_DELAY = 6000; // delay with server message via
	public static final int CLIENT_UID = 2856789;
											// in-game text box.

	/**
	 * URLS
	 */

	public static final String HOMEPAGE_URL = "www.destroscape.com";
	public static final String FORUM_URL = "www.destroscape.com/forum";
	public static final String VOTE_URL = "www.destroscape.com/vote";
	public static final String DONATE_URL = "https://www.paypal.com/uk/cgi-bin/webscr?cmd=_flow&SESSION=RrsWkePycitasNSFf78lbCvBSGTk036oL7nblMz4lZM_LRjwbvRHMOKkB_m&dispatch=5885d80a13c0db1f8e263663d3faee8d66f31424b43e9a70645c907a6cbd8fb4";
	public static final String SUGGESTIONS_URL = "http://destroscape.com/index.php?/forum/14-suggestions/";
	public static final String UPDATES_URL = "www.destroscape.com/index.php?/forum/9-updates/";
	public static final String HIGHSCORES_URL = "www.destroscape.com/Highscores";


	/**
	 * Choose exp
	 */
	public static final String XP_NOVICE = "novice";
	public static final String XP_EXPERT = "expert";
	public static final String XP_LEGEND = "legend";
	public static final String XP_NONE = "none";


	/**
	 * Sound
	 */

	public static final boolean SOUND = true;

	/** Client Config **/
	public static final int CLIENT_VERSION = 317; // our client build
	public static final int CONNECTION_DELAY = 500; // how long one ip can keep
													// connecting.
	public static final int IPS_ALLOWED = 1; // how many ips are allowed
	public static final int TIMEOUT = 20; //
	public static final int CYCLE_TIME = 600; // how long it takes to complete
												// one cycle (600 default)
	public static final int BUFFER_SIZE = 10000; //
	public static final int[] IMMEDIATE_PROCESS_PACKET_IDS = { // Need to
																// implement
			72, 131, // Attack NPC
			73, 249, // Attack PLAYER
			122, // Click ITEM
			185, // Clicking BUTTONS
			214, // Move ITEMS
			41, // Wear ITEM
			145, // Remove ITEM
	};
	
	/**
	 * Yelling System
	 */
	public static String[] BADTAGS = {
		"add",
		"banned",
		"tags",
		"yourself",
		":awesomeface:",
		"owner",
		"Owner",
		"Mod",
		"mod",
		"Moderator",
		"moderator",
		"Admin",
		"admin",
		"Administrator",
		"administrator"
	};
	
	/** Beta Mode TOGGLE **/
	public static boolean BETA_MODE = false;

	/** Player Config **/
	public static final int MAX_PLAYERS = 2001; // change height according to
												// average amount of players
												// online, Peak average.
	public static final int SAVE_TIMER = 3600; // save every 30 minute
	// public static final boolean ALLOWPINS = true; // turn bank pin system
	// on/off
	public static final int ITEM_LIMIT = 25000; // item ID limit.
	public static final int MAXITEM_AMOUNT = Integer.MAX_VALUE; // maximum
																// Amount of one
																// item a player
																// can have
	public static final int BANK_SIZE = 352; // size of bank
	// public static final int POT_DELAY = 120; // potion consumption delay
	public static final int FOOD_DELAY = 180; // food consumption delay
	public static final boolean PRAYER_POINTS_REQUIRED = true; // you need
																// prayer points
																// to use prayer
	public static boolean IDLE_LOGOUT = true; // Do players log out when
												// IDLEing?

	/** Player Spawn Config **/
	public static final int START_LOCATION_X = 3164; // X Co-ordinate of
														// starting location
	public static final int START_LOCATION_Y = 3480; // Y Co-ordinate of
														// starting location
	public static final int START_HEIGHT_LEVEL = 2; // height level of starting
													// location
	public static final int START_PLAYER_RIGHT = 0; // player right new players
													// have when they first log
													// in
	// public static boolean CHARACTER_CREATION = true; // do players select
	// their look when they first log-in?
	public static final int RESPAWN_X = Config.HOME_X;
	public static final int RESPAWN_Y = Config.HOME_Y;
	public static final int DUELING_RESPAWN_X = 3362; // duel X
	public static final int DUELING_RESPAWN_Y = 3263; // duel Y
	public static final int RANDOM_DUELING_RESPAWN = +Misc.random(5); // random																		// co-ordinates

	/** NPC Config **/
	public static final int NPC_RANDOM_WALK_DISTANCE = 15; // the square created
															// , 3x3 so npc
															// can't move out of
															// that box when
															// randomly walking
	public static final int NPC_FOLLOW_DISTANCE = 10; // how far can the npc
														// follow you from it's
														// spawn point,
	public static final int[] UNDEAD_NPCS = { 90, 91, 92, 93, 94, 103, 104, 73,
			74, 75, 76, 77 }; // undead npcs

	/** Combat **/
	public static final int NO_TELEPORT_WILD_LEVEL = 20; // level you can't tele
															// on and above
	public static final int SKULL_TIMER = 2400; // how long does the skull last?
												// seconds x 2
	public static final boolean SINGLE_AND_MULTI_ZONES = true; // multi and
																// single zones?
	public static final boolean COMBAT_LEVEL_DIFFERENCE = true; // wildy levels
																// and combat
																// level
																// differences
																// matters
	public static final boolean itemRequirements = true; // attack, def, str,
															// range or magic
															// levels required
															// to wield weapons
															// or wear items?
	public static final boolean PRAYER_LEVEL_REQUIRED = true; // need prayer
																// level to use
																// different
																// prayers

	public static final boolean CORRECT_ARROWS = true; // correct arrows for
														// bows?
	public static final boolean CRYSTAL_BOW_DEGRADES = true; // magic rune
																// required?
	public final static int[] BOWS = { 9185, 18357, 839, 845, 847, 851, 855,
			859, 841, 843, 849, 853, 857, 861, 4212, 4214, 4215, 11235, 15701,
			15702, 15703, 15704, 4216, 4217, 4218, 4219, 4220, 4221, 4222,
			4223, 6724, 4734, 4934, 4935, 4936, 4937 , 20171};
	public final static int[] ARROWS = { 882, 884, 886, 888, 890, 892, 4740,
			11212, 9140, 9141, 4142, 9143, 9144, 9240, 9241, 9242, 9243, 9244,
			9245 };
	public final static int[] NO_ARROW_DROP = { 4212, 4214, 4215, 4216, 4217,
			4218, 4219, 4220, 4221, 4222, 4223, 4734, 4934, 4935, 4936, 4937, 20171 };
	public final static int[] OTHER_RANGE_WEAPONS = { 863, 864, 865, 866, 867,
			868, 869, 806, 807, 808, 809, 810, 811, 825, 826, 827, 828, 829,
			830, 800, 801, 802, 803, 804, 805, 6522, 3093,11230 };

	public static final int TELEBLOCK_DELAY = 500; // how long does teleblock
													// last for.
	public static final boolean MAGIC_LEVEL_REQUIRED = true; // need magic level
																// to cast spell
	public static final int GOD_SPELL_CHARGE = 300000; // how long does god
														// spell charge last?
	public static final boolean RUNES_REQUIRED = true; // magic rune required?
	public final static int[] REDUCE_SPELLS = { 1153, 1157, 1161, 1542, 1543,
			1562 };
	public final static int[] REDUCE_SPELL_TIME = { // how long does the other
													// player stay immune to the
													// spell
			250000, 250000, 250000, 500000, 500000, 500000 };

	   public static final int[] NON_ATTAKABLE_NPCS = new int[] { 
	        8923, 599, 2286, 555, 3021, 700, 541, 545, 557, 461, 6971, 550, 
	        556, 1597, 2282, 211, 246, 494, 2566, 3789, 1696, 495, 700, 2258, 
	        6971, 1035, 5082, 5083, 6063, 1028, 5084, 5085
	    }; 	

/** Item Config **/
	// what items can't be sold in any store
	public static final int[] ITEM_SELLABLE = { 560, 565, 566, 561, 9075, 7406, 7405, 7404, 10328, 10329, 19308, 19309, 19310, 19311, 19312, 19313, 19314, 19315, 19316, 19317, 
			19318, 19319, 19320, 19321, 19322, 
			18346, 2363, 2364, 18357, 10330, 10331, 10332, 10333,
			10334, 10335, 10336, 10337, 10338, 10339, 10340, 10341, 10342, 10343, 10344, 10345,
			10346, 10347, 10348, 10349, 10350, 10351, 10352, 10353,
			19669, 18359, 18363, 18361, 18334, 18335, 19893, 18333, 18347,
			18349, 18351, 18353, 18355, 3842, 3844, 3840, 8844, 8845, 8846,
			8847, 8848, 8849, 8850, 20072, 10551, 6570, 7462, 7461, 7460, 7459,
			7458, 7457, 7456, 7455, 7454, 8839, 8840, 8842, 11663, 11664,
			13263, 11665, 10499, 9798, 9804, 9780, 9795, 9792, 9774, 9771,
			9777, 9786, 9810, 9765, 995, 9748, 9754, 9751, 9769, 9757, 9760,
			9763, 9802, 12169, 12171, 12170, 9808, 9784, 9799, 9805, 9781,
			9796, 9793, 9775, 9772, 9778, 9787, 9811, 9766, 9749, 9755, 9752,
			9770, 9758, 9761, 9764, 9803, 9809, 9785, 9800, 9806, 9782, 9797,
			9794, 9776, 9773, 9779, 9788, 9812, 9767, 9747, 9753, 9750, 9768,
			9756, 9759, 9762, 9801, 9807, 9783, 12158, 12159, 12160, 12161, 20747, 20767, 20751, 20768, 
			12162, 12163, 12164, 12165, 12166, 12167, 12168, 15492, 8839, 8840,
			8842, 11663, 11664, 12169, 12171, 11665, 10499, 15441, 15442,
			15443, 15444, 15701, 15702, 15703, 15704, 9747, 9753, 9750, 9768,
			9756, 9759, 9762, 9801, 9807, 9783, 9798, 9804, 9780, 9795, 9792,
			9774, 9771, 9777, 9786, 9810, 9765, 9948, 12169, 15086, 15088, 15090,
			15092, 15094, 15096, 15098, 15100, 19712 };
	// what items can't be traded or staked to players
	public static final int[] ITEM_TRADEABLE = { 7406, 7405, 7404, 10328, 10329, 9747, 9753, 9750, 9768, 9756,
			9759, 9762, 9801, 9807, 9783, 9798, 9804, 9780, 9795, 9792, 9774,
			9771, 9777, 9786, 9810, 9765, 9948, 12169, 18346, 18357, 19669,
			18359, 18363, 18361, 18334, 18335, 19893, 18333, 18347, 18349,
			18351, 18353, 18355, 3842, 3844, 3840, 8844, 8845, 8846, 8847,
			8848, 8849, 8850, 20072, 9798, 9804, 9780, 9795, 9792, 9774, 13263,
			9771, 9777, 9786, 9810, 9765, 7454, 8839, 8840, 8842, 11663, 11664,
			12169, 12171, 11665, 10499, 9748, 9754, 9751, 9769, 9757, 9760,
			9763, 9802, 9808, 9784, 9799, 9805, 9781, 9796, 9793, 9775, 9772,
			12170, 9778, 9787, 9811, 9766, 9800, 9806, 9782,
			9797, 9794, 9776, 9773, 9779, 9788, 9812, 9767, 9749, 9755, 9752,
			9770, 9758, 9761, 9764, 9803, 9809, 9785, 10551, 6570, 7462, 7461,
			7460, 7459, 7458, 7457, 7456, 7455, 9747, 9753, 9750, 9768, 9756,
			9759, 9762, 9801, 9807, 9783, 1508, 15492, 15441, 15442, 15443,
			15444, 15701, 15702, 15703, 15704, 9948, 12169, 14001, 14002, 20747, 20767, 20751, 20768, 
			14003, 14004, 14005, 14006 ,14007, 14008, 14009 ,14010, 15086, 15088, 15090, 15092, 15094,
			15096, 15098, 15100, 19712, 6570};
	//18778,
	//18779, 18780, 18781, 18782,
	// what items can't be dropped on ground
	public static final int[] UNDROPPABLE_ITEMS = { 7406, 7405, 7404, 10328, 10329, 1508, 15086, 15088, 15090, 15092, 
		15094, 15096, 15098, 15100, 7454, 7456, 7458, 7460, 7462, 6570, 20747, 20767, 20751, 20768,  };
	// fun weapons for dueling
	public static final int[] FUN_WEAPONS = { 2460, 2461, 2462, 2463, 2464,
			2465, 2466, 2467, 2468, 2469, 2470, 2471, 2471, 2473, 2474, 2475,
			2476, 2477 };

	/** Slayer Config **/
	public static final int[][] SLAYER_TASKS = { { 1, 87, 90, 4, 5 }, // low
																		// tasks
			{ 6, 7, 8, 9, 10 }, // med tasks
			{ 11, 12, 13, 14, 15 }, // high tasks
			{ 1, 1, 15, 20, 25 }, // low reqs
			{ 30, 35, 40, 45, 50 }, // med reqs
			{ 60, 75, 80, 85, 90 } // high reqs
	};

	public static final int[] QuestInterface = { 8145, 8147, 8148, 8149, 8150,
			8151, 8152, 8153, 8154, 8155, 8156, 8157, 8158, 8159, 8160, 8161,
			8162, 8163, 8164, 8165, 8166, 8167, 8168, 8169, 8170, 8171, 8172,
			8173, 8174, 8175, 8176, 8177, 8178, 8179, 8180, 8181, 8182, 8183,
			8184, 8185, 8186, 8187, 8188, 8189, 8190, 8191, 8192, 8193, 8194,
			8195, 12174, 12175, 12176, 12177, 12178, 12179, 12180, 12181,
			12182, 12183, 12184, 12185, 12186, 12187, 12188, 12189, 12190,
			12191, 12192, 12193, 12194, 12195, 12196, 12197, 12198, 12199,
			12200, 12201, 12202, 12203, 12204, 12205, 12206, 12207, 12208,
			12209, 12210, 12211, 12212, 12213, 12214, 12215, 12216, 12217,
			12218, 12219, 12220, 12221, 12222, 12223 };

	/** Experience Config **/
	/* EXP rates */
	public static double BRAWLER = 0.5; // XP bonuses
	public static double SERVER_EXP_BONUS = 1; // XP bonuses
	public static boolean doubleEXPWeekend = false;
	public static final int MELEE_EXP_RATE = 1000; // damage * exp rate.
	public static final int RANGE_EXP_RATE = 900; // damage * exp rate.
	public static final int MAGIC_EXP_RATE = 900; // damage * exp rate.

	/** Teleport Config **/
	// amulet of glory
	public static final int HOME_X = 3164;
	public static final int HOME_Y = 3480;

	public static final int EDGEVILLE_X = 3087; // X
	public static final int EDGEVILLE_Y = 3500; // Y
	public static final String EDGEVILLE = ""; // name of teleport
	public static final int AL_KHARID_X = 3293; // X
	public static final int AL_KHARID_Y = 3174; // Y
	public static final String AL_KHARID = ""; // name of teleport
	public static final int KARAMJA_X = 2956; // X
	public static final int KARAMJA_Y = 3146; // Y
	public static final String KARAMJA = ""; // name of teleport
	public static final int MAGEBANK_X = 2538; // X
	public static final int MAGEBANK_Y = 4716; // Y
	public static final String MAGEBANK = ""; // name of teleport
	// ring of dueling
	// modern
	public static final int VARROCK_X = 3210;
	public static final int VARROCK_Y = 3424;
	public static final String VARROCK = "";
	public static final int LUMBY_X = 3222;
	public static final int LUMBY_Y = 3218;
	public static final String LUMBY = "";
	public static final int FALADOR_X = 2964;
	public static final int FALADOR_Y = 3378;
	public static final String FALADOR = "";
	public static final int CAMELOT_X = 2757;
	public static final int CAMELOT_Y = 3477;
	public static final String CAMELOT = "";
	public static final int ARDOUGNE_X = 2662;
	public static final int ARDOUGNE_Y = 3305;
	public static final String ARDOUGNE = "";
	public static final int WATCHTOWER_X = 3087;
	public static final int WATCHTOWER_Y = 3500;
	public static final String WATCHTOWER = "";
	public static final int TROLLHEIM_X = 3243;
	public static final int TROLLHEIM_Y = 3513;
	public static final String TROLLHEIM = "";
	// ancient
	public static final int PADDEWWA_X = 3098;
	public static final int PADDEWWA_Y = 9884;
	public static final int SENNTISTEN_X = 3322;
	public static final int SENNTISTEN_Y = 3336;
	public static final int KHARYRLL_X = 3492;
	public static final int KHARYRLL_Y = 3471;
	public static final int LASSAR_X = 3006;
	public static final int LASSAR_Y = 3471;
	public static final int DAREEYAK_X = 3161;
	public static final int DAREEYAK_Y = 3671;
	public static final int CARRALLANGAR_X = 3156;
	public static final int CARRALLANGAR_Y = 3666;
	public static final int ANNAKARL_X = 3288;
	public static final int ANNAKARL_Y = 3886;
	public static final int GHORROCK_X = 2977;
	public static final int GHORROCK_Y = 3873;
	// lunar

	/** Player - Item Transfer Options **/
	// Regular Player
	public static final boolean REGULAR_CAN_TRADE_ITEMS = true;
	public static final boolean REGULAR_CAN_SELL_ITEMS = true;
	public static final boolean REGULAR_DROP_ITEMS = true;
	// Donator
	public static final boolean DONATOR_CAN_TRADE_ITEMS = true;
	public static final boolean DONATOR_CAN_SELL_ITEMS = true;
	public static final boolean DONATOR_DROP_ITEMS = true;
	// Super Donator
	public static final boolean SUPDONATOR_CAN_TRADE_ITEMS = true;
	public static final boolean SUPDONATOR_CAN_SELL_ITEMS = true;
	public static final boolean SUPDONATOR_DROP_ITEMS = true;
	// Moderator
	public static final boolean MODERATOR_CAN_TRADE_ITEMS = true;
	public static final boolean MODERATOR_CAN_SELL_ITEMS = true;
	public static final boolean MODERATOR_DROP_ITEMS = true;
	// Developer
	public static final boolean DEVELOPER_CAN_TRADE_ITEMS = true;
	public static final boolean DEVELOPER_CAN_SELL_ITEMS = true;
	public static final boolean DEVELOPER_DROP_ITEMS = true;
	// Co-Onwer
	public static final boolean COOWNER_CAN_TRADE_ITEMS = true;
	public static final boolean COOWNER_CAN_SELL_ITEMS = true;
	public static final boolean COOWNER_DROP_ITEMS = true;
	// Owner
	public static final boolean OWNER_CAN_TRADE_ITEMS = true;
	public static final boolean OWNER_CAN_SELL_ITEMS = true;
	public static final boolean OWNER_DROP_ITEMS = true;
	public static final int INCREASE_SPECIAL_AMOUNT = 30000;
	
	public static final boolean PARTYROOM_DROP_DELAY = false;
	public static final boolean PARTYROOM_DISABLED = false;
	public static final boolean PARTYROOM_ADMINS_ONLY = true;
	
	public static final int[] ITEMS_KEPT_ON_DEATH = {15086,15088,15090,15092,15094,15096,15098,15100};
	
	public static final boolean DUNGEONEERING_ACCESSIBLE = true;

}
